packages/eslint-plugin/docs/rules/no-explicit-any.md

```diff
@@ -158,7 +158,24 @@ interface Garply {
 
 ## When Not To Use It
 
-If an unknown type or a library without typings is used
+**Higher Order Functions**. If you're typing a higher order function such as `compose`, `pipe`, etc, you may need `any` because TypeScript types are not fully co-expressive with idiomatic functional JavaScript. Such functions are intentionally generic, and you can assign explicit types later to the returned functions, e.g.:
```

Comment:
> **Higher Order Functions**

Functional shenanigans are a common example of type system edge cases, yes, but they're not the only one. And -similar to my other comment- functional terminology can be off-putting when phrased as a top-level reason rather than an example of a more approachable term. How about we make this more general?

```diff
- **Higher Order Functions**. etc. etc.
+ **Difficult-to-represent TypeScript types**.
+ Some code patterns can be difficult to represent exactly in the TypeScript type system, and/or expose legitimate edge cases for using `any`.
+ For example, functional programming concepts such as composing and piping sometimes necessitate using `any`.
```

Thoughts?

Reply:
Love it.